controllers/config.go

```diff
@@ -78,6 +81,28 @@ func ParseOdfConfigMapRecords(logger logr.Logger, configmap corev1.ConfigMap, fn
 			continue
 		}
 
+		if record.Namespace == "" {
+			record.Namespace = OperatorNamespace
+		}
+
 		fn(&record, key, value)
 	}
 }
+
+func GetTargetNamespaces(logger logr.Logger, configmap corev1.ConfigMap) []string {
+
+	var namespacesMap = make(map[string]bool)
+	var namespaces []string
+
+	ParseOdfConfigMapRecords(logger, configmap, func(record *OdfOperatorConfigMapRecord, key, rawValue string) {
+
+		if !namespacesMap[record.Namespace] {
+			namespaces = append(namespaces, record.Namespace)
+		}
+		namespacesMap[record.Namespace] = true
+	})
+
+	logger.Info("namespace records", "namespaces", namespaces)
+
+	return namespaces
```

Comment:
There is no need to have both a map and a list, use map only and construct the list only when your map is full. This will sidestep all of the potential allocations that comes with `append` (`append` can potentially copy the entire map for a resize)

```suggestion
	ParseOdfConfigMapRecords(logger, configmap, func(record *OdfOperatorConfigMapRecord, key, rawValue string) {
		namespacesMap[record.Namespace] = true
	})

	namespaces := slices.Collect(maps.Keys(namespaceMap))
	logger.Info("namespace records", "namespaces", namespaces)
	return namespaces
```

Reply:
Done